cli/js/web/storage.ts

```diff
@@ -0,0 +1,60 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+import * as domTypes from "./dom_types.d.ts";
+
+class Storage implements domTypes.Storage {
+  private data: Map<string, string> = new Map();
```

Comment:
shouldnt this be using the ES class private field indentifier (`#`) opposed to the TS `private` keyword?

Reply:
Actually `Proxy` doesn't deal well with ES private fields... But I should use a `Symbol` instead to disallow access to the internal `Map` from userland.

Reply:
ES Private fields and Proxies interact exactly as intended, in the sense that ES private fields are private.  The fact they are not accessible via the proxy or observable is exactly the behaviour desired.  We should use ES private fields where possible.  (when dealing with specs and internal slots that are accessible outside of the class in the "internal" spec code, symbols are appropriate).  If data is not externally observable, which it shouldn't be, it should be a ES Private field.

Reply:
The issue is that when `Reflect.get` returns the `Storage` method, it gets scoped to the `Proxy` object, and the following error is thrown: `TypeError: Cannot read private member #data from an object whose class did not declare it.`. I came up with a workaround but I don't find it very elegant:

```typescript
  get(target: domTypes.Storage, key: string): string | null | number | void {
    if ("function" === typeof target[key]) {
      return target[key].bind(target);
    } else if (key === "length") {
      return target.length;
    } else {
      return target.getItem(key);
    }
  },
```

Maybe I'm missing something obvious and there is a simpler way of implementing that?